homeassistant/components/poolstation/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""The Poolstation integration."""
+from datetime import timedelta
+import logging
+
+import aiohttp
+from pypoolstation import Account, Pool
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import COORDINATORS, DEVICES, DOMAIN, TOKEN
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL = timedelta(seconds=30)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Poolstation from a config entry."""
+    account = Account(token=entry.data[TOKEN])
+
+    try:
+        pools = await Pool.all(account=account)
+    except aiohttp.ClientError as err:
+        raise ConfigEntryNotReady from err
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        COORDINATORS: {},
+        DEVICES: {},
+    }
+
+    for pool in pools:
+        pool_id = pool.id
+        coordinator = PoolstationDataUpdateCoordinator(hass, pool)
+        await coordinator.async_config_entry_first_refresh()
+
+        hass.data[DOMAIN][entry.entry_id][DEVICES][pool_id] = pool
+        hass.data[DOMAIN][entry.entry_id][COORDINATORS][pool_id] = coordinator
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class PoolstationDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching Poolstation device info."""
+
+    def __init__(self, hass: HomeAssistant, pool: Pool) -> None:
+        """Initialize global Poolstation data updater."""
+        self._device = pool
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}-{pool.id}",
```

Comment:
The DataUpdateCoordinator name is used in the DataUpdateCoordinator logs. The pool name may be clearer to users than the pool id. `DOMAIN` is already in the logger and can optionally be omitted from the coordinator name.

Reply:
Yep, the id is a number and it won't be very helpful. Even if only a very tiny fraction of people will have more than one pool reallistically.